### PR TITLE
Fix2 for express compression module not activating

### PR DESCRIPTION
**Pull request by hdf**
Fix for Issues:
#199
and
expressjs/compression#38

### CODE REVIEW

**Comment by indutny** [comment]
Yay, looking fantastic! :) Btw, do you know that you could just push to the same branch and the PR will automatically update with a new commits (even on a force push)? ;)

Now comes the second part of any change ;) It needs to have a test. To keep things simple I'd just copy on of the tests there: https://github.com/indutny/node-spdy/blob/master/test/unit/connect-test.js and add the headers test to it. May I ask you to do it?

**Comment by hdf** [comment]
Is this what you meant? Is it not a problem, that the `req.rawHeaders` will not be modified this way?

**Comment by indutny** [comment]
Huuuh? Impossible!

**Comment by hdf** [comment]
Ah, I see, I don't need to set `rawHeaders` if I set `state.headers['accept-encoding']` in stream.js. Although I don't understand why, but it is clear, that that is the better place for this snippet now. :)

**Comment by hdf** [comment]
Is it OK like this?

File: test/unit/connect-test.js [hunk 1]
```diff
@@ -202,4 +209,29 @@ suite('A SPDY Server / Connect', function() {
       done();
     });
   });
+
+  test('should add accept-encoding header to request headers, if none present', function(done) {
```
Comment (indutny) [inline test/unit/connect-test.js]: 80 column limit, please.

**Comment by indutny** [comment]
One minor nit, otherwise LGTM!

**Comment by hdf** [comment]
Github editor does not tell me where the 80 columns end.. :(

File: lib/spdy/stream.js [hunk 2]
```diff
@@ -50,6 +50,13 @@ function Stream(connection, options) {
   state.id = options.id;
   state.associated = options.associated;
   state.headers = options.headers || {};
+
+  // Protocol standard compliance:
+  // Client does not have to send this header, we must assume,
+  // that it can handle compression
+  if (connection._spdyState.isServer && !state.associated && !state.headers['accept-encoding'])
```
Comment (indutny) [inline lib/spdy/stream.js]: Sorry, here too ;)

File: test/unit/connect-test.js [hunk 3]
```diff
@@ -202,4 +209,30 @@ suite('A SPDY Server / Connect', function() {
       done();
     });
   });
+
+  test('should add accept-encoding header to request headers, if none present',
```
Comment (indutny) [inline test/unit/connect-test.js]: I'd rather split the string itself.

**Comment by hdf** [comment]
It is exactly 79 characters long now. :) You sure?

**Comment by hdf** [comment]
Maybe I should write the version bump as well in the `package.json`? :)

File: test/unit/connect-test.js [hunk 4]
```diff
@@ -202,4 +209,30 @@ suite('A SPDY Server / Connect', function() {
       done();
     });
   });
+
+  test('should add accept-encoding header to request headers, \
```
Comment (indutny) [inline test/unit/connect-test.js]: Ahhah, ok. I'll fix it for you.

**Comment by indutny** [comment]
@hdf landed in https://github.com/indutny/node-spdy/commit/a9a8069592f3c36b82b22518aa6613f77311cb98, thank you!

**Pull request closed without merging**